web/app/scripts/views/map/layers-controller.js

```diff
@@ -203,8 +204,19 @@
             });
 
             // user-uploaded boundary layer(s)
-            var boundaryLayerOptions = angular.extend(defaultLayerOptions, {zIndex: 2});
-            var boundaryLayer = new L.tileLayer(boundaryUrl, boundaryLayerOptions);
+            var availableBoundaries = $q.defer();
+            GeographyState.getOptions().then(function(boundaries) {
+                var _overlays = {};
+                var boundaryLayerOptions = angular.extend(defaultLayerOptions, {zIndex: 2});
+                boundaries.map(function(boundary) {
```

Comment:
This can probably be simplified a little bit by having the map return pairs of [label, boundaryLayer] and calling _.zipObject on it, rather than appending to the _overlays variable.